src/utils/useGetReplaceMessageGasParams.ts

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from 'react'
+import { LotusMessage } from '@glif/filecoin-message'
+import { FilecoinNumber } from '@glif/filecoin-number'
+import Filecoin from '@glif/filecoin-wallet-provider'
+
+interface ProviderGasParams {
+  gasFeeCap: string
+  gasPremium: string
+  gasLimit: number
+}
+
+interface GasParams {
+  gasFeeCap: FilecoinNumber
+  gasPremium: FilecoinNumber
+  gasLimit: FilecoinNumber
+}
+
+interface UseGetReplaceMessageGasParamsResult {
+  gasParams: GasParams
+  loading: boolean
+  error: Error
+}
+
+export const useGetReplaceMessageGasParams = (
+  provider: Filecoin,
+  message?: LotusMessage,
+  minimum?: boolean
+): UseGetReplaceMessageGasParamsResult => {
+  const [gasParams, setGasParams] = useState<GasParams | null>(null)
+  const [loading, setLoading] = useState<boolean>(false)
+  const [error, setError] = useState<Error | null>(null)
+
+  useEffect(() => {
+    const getGasParams = minimum
+      ? provider.getReplaceMessageMinGasParams
+      : provider.getReplaceMessageGasParams
+    setGasParams(null)
+    setError(null)
+    if (message) {
```

Comment:
I think here we should also check if we're not already loading.

It feels like something (looking at the `provider` dep in the useEffect dependency array) could trigger this `useEffect` hook to rerender unexpectedly, which could cause some trouble when we keep resetting gasParams in state.

Reply:
I wonder, if the message does change while loading and we ignore it, we might end up with old / wrong data? Maybe this doesn't apply in the current use case of tx replacement, but it doesn't feel completely right.

Do you think that resetting the gas params could cause the provider to change?

Reply:
Why would the message change while loading, and even if it did change, why would we ignore it here?

A back to back message change (i would think) would just cause the `setGasParams` to be set again with the updated gas params?

I foresee a weird series of events where the `provider` could change, mostly around the way the Ledger Device wallet subprovider works. We basically have to reinstantiate a Ledger wallet subprovider instance in certain circumstances, which i think would trigger a falsey equality check by react's dependency array checker

Reply:
Curious what you think about this:
https://github.com/glifio/react-components/pull/282